docs/design/windows-images.md

```diff
@@ -0,0 +1,164 @@
+
+Teams needing to create and test prerelease patches will need to use a private branch and Azure DevOps Pipeline to avoid unintentional security disclosures. This should be kept in-sync with the public repo but use a different Azure subscription and accounts.
+
+### AKS-Users testing new Windows patches
+
+During the VHD build, `WindowsPatchList` and `WindowsPrivatePatchList` control what cumulative updates and/or patches are installed. However, there may be cases where a private patch has been given to a customer who cannot build a new VHD. The existing [windows-patches](C:\Users\patri\Source\aks-engine\extensions\windows-patches) extension should still work to do this.
+
+
+### Testing new runtime component versions in AKS-Engine
+
+Testing dependencies such as Docker, ContainerD, CNI plugins, ... that are released through AKS-Engine builds should follow the same process and cadence as the Linux VHD workflow.
+
+### Testing upstream builds of Kubernetes
+
+The main [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) codebase is tested on Azure using AKS-Engine. The test results are not triggered by or reported to the AKS-Engine repo. This is done by setting the `kubernetesConfig.WindowsNodeBinariesURL` in the API model. That will be downloaded to each node and overwrite what was installed in the VHD.
+
+### Testing upstream builds of CRI-ContainerD
+
+Similar to kubernetes/kubernetes, containerd/containerd needs a working deployment on Azure to test new builds. That CI system can produce a ContainerD build, then set `kubernetesConfig.WindowsContainerdURL` in the API model to get a working deployment.
+
+### Testing upstream builds of CNI plugins
+
+Similar to kubernetes/kubernetes, [Microsoft/SDN](https://github.com/Microsoft/SDN) needs a working deployment on Azure to test new builds. That CI system can produce a ContainerD build, then set `kubernetesConfig.WindowsSdnPluginURL` in the API model to get a working deployment.
+
+## Workflow
+
+### VHD Build
+
+These configurations drive the VHD build process. As such, they should be version controlled in source code for reproducability. They will be set as variables for a Packer build. Some of these have intentional overlap between the VHD build steps and node deployment to fulfill different use cases.
+
+1. (optional, only if installing from ISO) Run Packer with an unattend file to get a working Windows setup from an ISO image
+  * Install optional features: `Containers`, `Hyper-V`
+  * Enable remote access with WinRM & Remote Desktop
+  * Install the [Azure Agent](https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/agent-windows#manual-installation)
+  * Reboot
+1. Build a cache of all needed dependencies - Docker EE, Moby, ContainerD, Azure-CNI, SdnBridge, and Kubernetes

```

Comment:
Since on Azure Stack we don't have Azure-CNI, We might need to package Kubenet/flannel.